services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/client/gRPC/GoalStateClientImpl.java

```diff
@@ -54,16 +53,17 @@
 import io.opentracing.contrib.grpc.TracingClientInterceptor;
 
 @Service("grpcGoalStateClient")
+@ComponentScan(value = "com.futurewei.alcor.netwconfigmanager.cache")
 public class GoalStateClientImpl implements GoalStateClient {
-
-    private static GoalStateClientImpl instance = null;
-
     private static final Logger logger = LoggerFactory.getLogger();
 
     private int hostAgentPort;
 
     private final ExecutorService executor;
 
+    @Autowired
+    private ResourceStateCache resourceStateCache;
+
```

Comment:
@DavidLiu506 I don't think gRPC client is a good place to read/write a cache, which is usually supposed to be done in upper layer.

Reply:
Good suggestions.  Thanks!